src/java6/java/org/spongepowered/launch/JavaVersionCheckUtils.java

```diff
@@ -51,6 +56,19 @@ public static void ensureJava8() {
             System.out.println("You may be running an outdated version of Java. Any crashes from Sponge may require an update to Java.");
         }
     }
+    
+    public static void checkForNewVersions() {
+        String version = getCurrentVersion();
+        if (getVersionValue(version) > getVersionValue(REQUIRED_VERSION)) {
+            String error = String.format(UNSUPPORTED, version, REQUIRED_VERSION);
+
+                if (!GraphicsEnvironment.isHeadless()) {
+                    JOptionPane.showMessageDialog(null, error, "PEBKACException!", JOptionPane.ERROR_MESSAGE);
+                }
+            throw new RuntimeException(error);
```

Comment:
You've removed the if but left the print after the throw, so this doesn't compile.